doc/CB-to-STH-documentation/TourGuide-STH-Documentation.md

```diff
@@ -0,0 +1,227 @@
+To query all entities that have been created we need to perform a GET operation on the following destination (the query should retrieve all previous entities we added previously)
+
+> localhost:1026/v2/entities
+
+![Select All entities](Images/selectAllEntities.png)
+
+
+
+## Operations on Attributes
+Another type of modifications can be done on attributes as individuals rather than the whole entity, these operations can be done on attributes. Such operations are; get attribute, put attribute, get value of an attribute, put value for specific attribute, and delete an attribute. 
+
+### Modify an Attribute
+In our case we want to modify the attribute value for the Car1 entity, this can be done by performing a PUT request to the following destination with the same 3 headers parameters we passed previously
+
+> localhost:1026/v2/entities/Car1/attrs/speed/value
+
+![modify attribute's value](Images/modifyAValue.png)
+
+## Create Subscriptions
+
+Context Consumers can subscribe to receive context information that satisfy certain conditions using the subscribe operation. Such subscriptions may have an expiration time. In return, the Context Broker notifies updates on context information to subscribed Context Consumers by invoking the notify operation they export.
+
+The _entities_ and _notifications_ subfields define the contents of the notification messages. 
+
+* "idPattern" is used to filter which entities we want. Its value is a regular expression. In our example we set it to retrieve all.
+* The URL where to send notifications is defined with the  url sub-field. Here we specified the URL of the STH-Comet.
+* "AttrsFormat": we set it to legacy because the STH only understands NGSI v1 notification payloads.
+* Subscriptions may have an expiration date (expires field), specified using the ISO 8601 standard format. Once subscription overpass that date, the subscription is simply ignored.
+
+> localhost:1026/v2/subscriptions
+
+![Add Subscription](Images/CreateSubscription.png)
+
+After creating a subscription we need to update the attribute values the same way we did before to be able to query some aggregations using the STH API. To see changes we need to update the attribute value in different times, the following example shows how we can change an attribute value several times
+
+![updating multiple values](Images/modifyValuesSeveralTimes.png)
+
+## Query Time Series Values
+Here we will query, through the STH, all the raw data for the values that have been changed. therefore we need to perform GET operation for the following destination:
+
+> localhost:8666/STH/v1/contextEntities/type/Car/id/Car1/attributes/speed?lastN=20
+
+![STH Raw Data](Images/QueryTimeSeriesValues.png)
+
+
+In the response payload we can see that the name of the changed attribute is "speed", followes with the values that have been changed within different time periods. For Exmple; we can see that the speed value 22 is happened on "2018-03-05T12:00:56:728Z".
```

Comment:
follows

Reply:
for example, we can see that the timestamp corresponding to the `speed` value of 22 is "2018-03-05T12:00:56:728Z"